File: app/src/main/java/com/qust/app/MainActivity.java
package com.qust.app;

import android.content.Intent;
import android.os.Bundle;

import com.qust.assistant.R;
import com.qust.assistant.util.SettingUtil;
import com.qust.base.ui.QFragmentActivity;
import com.qust.model.AutoQueryModel;
import com.qust.utils.UpdateUtils;

public class MainActivity extends QFragmentActivity{
	
	@Override
	protected void onCreate(Bundle paramBundle){
		super.onCreate(paramBundle);
		
		// 第一次使用跳转到引导页
		if(SettingUtil.getBoolean(getString(R.string.isFirstUse), true)){
			startActivity(new Intent(this, GuideActivity.class));
			return;
		}
		
		// 检查更新
		UpdateUtils.checkUpdateAsync(this);
		
		// 自动检查工具
		AutoQueryModel.startAutoQuery(this);
	}

	@Override
	public void startActivity(Intent intent){
		super.startActivity(intent);
		overridePendingTransition(R.anim.anim_right_in, 0);
	}
}

File: composeApp/src/androidMain/kotlin/com/qust/helper/ui/activity/MainActivity.kt
package com.qust.helper.ui.activity

import android.content.Intent
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import com.qust.helper.data.Keys
import com.qust.helper.ui.page.HomePage
import com.qust.helper.ui.theme.AppTheme
import com.qust.helper.utils.SettingUtils

class MainActivity: ComponentActivity() {

	val page = HomePage

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)

		// 第一次使用跳转到引导页（用户协议）
		if(SettingUtils[Keys.IS_FIRST_USE, true]) {
			startActivity(Intent(this, GuideActivity::class.java))
		}

		setContent { AppTheme { HomePage.ComposePage() } }
	}
}
